plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergNodePartitioningProvider.java

```diff
@@ -14,30 +14,63 @@
 package io.trino.plugin.iceberg;
 
 import com.google.inject.Inject;
+import io.trino.spi.NodeManager;
 import io.trino.spi.connector.BucketFunction;
+import io.trino.spi.connector.ConnectorBucketNodeMap;
 import io.trino.spi.connector.ConnectorNodePartitioningProvider;
 import io.trino.spi.connector.ConnectorPartitioningHandle;
 import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorSplit;
 import io.trino.spi.connector.ConnectorTransactionHandle;
 import io.trino.spi.type.Type;
 import io.trino.spi.type.TypeManager;
 import io.trino.spi.type.TypeOperators;
 import org.apache.iceberg.Schema;
 
 import java.util.List;
+import java.util.Optional;
+import java.util.function.ToIntFunction;
 
 import static io.trino.plugin.iceberg.IcebergUtil.schemaFromHandles;
 import static io.trino.plugin.iceberg.PartitionFields.parsePartitionFields;
+import static io.trino.spi.connector.ConnectorBucketNodeMap.createBucketNodeMap;
 
 public class IcebergNodePartitioningProvider
         implements ConnectorNodePartitioningProvider
 {
     private final TypeOperators typeOperators;
+    private static final int PARTITIONED_BUCKETS_PER_NODE = 32;
+
+    private final NodeManager nodeManager;
 
     @Inject
-    public IcebergNodePartitioningProvider(TypeManager typeManager)
+    public IcebergNodePartitioningProvider(TypeManager typeManager, NodeManager nodeManager)
     {
         this.typeOperators = typeManager.getTypeOperators();
+        this.nodeManager = nodeManager;
+    }
+
+    @Override
+    public Optional<ConnectorBucketNodeMap> getBucketNodeMapping(
```

Comment:
Can you explain why implementing this function is needed ?
Is there a reason to care about which node the engine chooses to schedule a bucket on, as long as it schedules all splits for a bucket on the same node ?
The big comment below is picked up from hive connector code where we added this code to parallelize writes to hive tables with partitioning and bucketing. I'm not sure that's relevant to this PR.